src/bucket/BucketTests.cpp

```diff
@@ -835,6 +835,59 @@ TEST_CASE("bucket persistence over app restart", "[bucket][bucketpersist]")
     }
 }
 
+TEST_CASE("bucket count", "[bucket][count]")
```

Comment:
You are testing sizeOfCurr new methods of BucketList.cpp here, what do you think about adding tests just for them? Make a bunch of tuples {ledger, level, expectedSizeOfCurr, expectedSizeOfSnap, expectedOldestLedgerInCurr, expectedOldestLedgerInSnap} and so on?

Reply:
I can definitely try to do this, but those functions are quite complicated and defined recursively so I would have to figure out how to evaluate them without using the implementations. Perhaps an easier idea is to try to write a test case which validates some invariants among those formulae, such as `oldestLedgerInSnap(ledger, level) + sizeOfSnap(ledger, level) == oldestLedgerInCurr(ledger, level)`. I could also add tests like `sizeOfSnap(ledger, level) == levelHalf(level)` implies `sizeOfSnap(ledger2, level) == levelHalf(level)` for `ledger2 > ledger`. Let me know if you think these kinds of tests are sufficient or if you would prefer me try to do some direct calculations.

Reply:
What about invariants and some edge cases?